### PR TITLE
luminous: rgw: conditionally allow builtin users with non-unique email addresses

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/40506

### CODE REVIEW

**Comment by smithfarm** [comment]
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_user.cc: In member function ‘int RGWUser::init(RGWUserAdminOpState&)’:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_user.cc:1792:27: error: request for member ‘get_val’ in ‘((RGWUser*)this)->RGWUser::store->RGWRados::ctx()->CephContext::_conf’, which is of pointer type ‘md_config_t*’ (maybe you meant to use ‘->’ ?)
   if (store->ctx()->_conf.get_val<bool>("rgw_user_unique_email")) {
                           ^
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_user.cc:1792:35: error: expected primary-expression before ‘bool’
   if (store->ctx()->_conf.get_val<bool>("rgw_user_unique_email")) {
                                   ^
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_user.cc:1792:35: error: expected ‘)’ before ‘bool’
src/rgw/CMakeFiles/rgw_a.dir/build.make:1962: recipe for target 'src/rgw/CMakeFiles/rgw_a.dir/rgw_user.cc.o' failed
make[3]: *** [src/rgw/CMakeFiles/rgw_a.dir/rgw_user.cc.o] Error 1
```

File: src/rgw/rgw_user.cc [hunk 1]
```diff
@@ -1789,9 +1789,11 @@ int RGWUser::init(RGWUserAdminOpState& op_state)
     found = (rgw_get_user_info_by_uid(store, user_id, user_info, &op_state.objv) >= 0);
     op_state.found_by_uid = found;
   }
-  if (!user_email.empty() && !found) {
-    found = (rgw_get_user_info_by_email(store, user_email, user_info, &op_state.objv) >= 0);
-    op_state.found_by_email = found;
+  if (store->ctx()->_conf->rgw_user_unique_email) {
```
Comment (cbodley) [inline src/rgw/rgw_user.cc]: i think the correct resolution here should be:
```diff
-  if (store->ctx()->_conf.get_val<bool>("rgw_user_unique_email")) {
+  if (store->ctx()->_conf->get_val<bool>("rgw_user_unique_email")) {

Comment (smithfarm) [reply src/rgw/rgw_user.cc]: Thanks

**Comment by yuriw** [comment]
[wip-yuri6-testing-2019-07-18-1538-luminous](https://trello.com/c/PDHX4Rvo/669-wip-yuri6-testing-2019-07-18-1538-luminous)

**Pull request merged**